test/integration/helpers_test.go

```diff
@@ -0,0 +1,330 @@
+//go:build integration
+
+package integration_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/neo4j/mcp/internal/database"
+	"github.com/neo4j/mcp/internal/tools"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j"
+	testcontainers "github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// TestContext holds common test dependencies
+type TestContext struct {
+	Ctx     context.Context
+	T       *testing.T
+	TestID  string
+	Service database.Service
+	Deps    *tools.ToolDependencies
+}
+
+// makeTestID returns a unique test id suitable for tagging resources created by tests.
+func makeTestID() string {
+	return fmt.Sprintf("test-%s", uuid.NewString())
+}
+
+func createNeo4jContainer(ctx context.Context) (testcontainers.Container, string, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        "neo4j:5.24.2-community",
+		ExposedPorts: []string{"7687/tcp"},
+		Env: map[string]string{
+			"NEO4J_AUTH":        "neo4j/password",
+			"NEO4JLABS_PLUGINS": `["apoc","graph-data-science"]`,
```

Comment:
I would accept some of these config/env as configuration argument - the immediate reason I have in mind is to test case where we want to test how we behave with missing optional requirements such as GDS.

Reply:
Sure!